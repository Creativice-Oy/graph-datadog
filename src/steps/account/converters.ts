import { User } from '@datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/User';
import { createIntegrationEntity } from '@jupiterone/integration-sdk-core';
import { Entities } from '../constants';

function getAccountKey(id: string): string {
  return `datadog_account:${id}`;
}

export function createAccountEntity(account: User) {
  return createIntegrationEntity({
    entityData: {
      source: account,
      assign: {
        _class: Entities.ACCOUNT._class,
        _type: Entities.ACCOUNT._type,
        _key: getAccountKey(account.email?.toString() as string),
        name: account.email,
        username: account.handle,
        email: account.email,
        id: account.email,
        verified: account.verified,
        disabled: account.disabled,
        icon: account.icon,
        displayName: account.name || '',
        accessRole: account.accessRole,
      },
    },
  });
}
