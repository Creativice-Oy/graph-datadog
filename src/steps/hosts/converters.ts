import { Host } from '@datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/Host';
import {
  createDirectRelationship,
  createIntegrationEntity,
  Entity,
  parseTimePropertyValue,
  Relationship,
  RelationshipClass,
} from '@jupiterone/integration-sdk-core';

import { Entities } from '../constants';

export function getHostKey(id: string): string {
  return `datadog_host:${id}`;
}

export function createHostEntity(host: Host): Entity | null {
  return createIntegrationEntity({
    entityData: {
      source: host,
      assign: {
        _type: Entities.HOST._type,
        _class: Entities.HOST._class,
        _key: getHostKey(host.id as unknown as string),
        id: `${host.id}`,
        aliases: host.aliases,
        apps: host.apps,
        awsName: host.awsName,
        hostname: host.hostName,
        isMuted: host.isMuted,
        lastReportedTime: parseTimePropertyValue(host.lastReportedTime),
        muteTimeout: host.muteTimeout,
        name: host.name,
        sources: host.sources,
        state:
          host.up
          ? 'running'
          : 'stopped',
      },
    },
  });
}

export function createAccountHostRelationship(
  account: Entity,
  host: Entity,
): Relationship {
  return createDirectRelationship({
    _class: RelationshipClass.HAS,
    from: account,
    to: host,
  });
}

