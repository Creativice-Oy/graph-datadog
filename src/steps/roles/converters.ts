import { Role } from '@datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/Role';
import {
  createIntegrationEntity,
  Entity,
  parseTimePropertyValue,
} from '@jupiterone/integration-sdk-core';

import { Entities } from '../constants';

function getRoleKey(id: string): string {
  return `datadog_role:${id}`;
}

export function createRoleEntity(role: Role): Entity | null {
  return createIntegrationEntity({
    entityData: {
      source: role,
      assign: {
        _type: Entities.ROLE._type,
        _class: Entities.ROLE._class,
        _key: getRoleKey(role.id as string),
        id: role.id,
        name: role.attributes?.name,
        userCount: role.attributes?.userCount,
        createdAt: parseTimePropertyValue(role.attributes?.createdAt),
        modifiedAt: parseTimePropertyValue(role.attributes?.modifiedAt),
      },
    },
  });
}
