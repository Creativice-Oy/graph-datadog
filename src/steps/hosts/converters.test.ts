import { getMockHost } from '../../../test/mocks';
import { createHostEntity } from './converters';

describe('#createHostEntity', () => {
  test('should convert to entity', () => {
    expect(createHostEntity(getMockHost())).toMatchSnapshot();
  });
});
