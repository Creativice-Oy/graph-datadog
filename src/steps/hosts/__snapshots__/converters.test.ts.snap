// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createHostEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "Host",
  ],
  "_key": "datadog_host:6264691456",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "aliases": Array [
          "Dhaiwats-MacBook-Air.local",
        ],
        "apps": Array [
          "ntp",
          "agent",
        ],
        "hostName": "Dhaiwats-MacBook-Air.local",
        "id": 6264691456,
        "isMuted": false,
        "lastReportedTime": 1657376703,
        "metrics": Object {
          "cpu": 5.1627164,
          "iowait": 0,
          "load": 0.1438288,
        },
        "muteTimeout": undefined,
        "name": "Dhaiwats-MacBook-Air.local",
        "sources": Array [
          "agent",
        ],
        "tagsBySource": Object {
          "Datadog": Array [
            "host:Dhaiwats-MacBook-Air.local",
          ],
        },
        "up": true,
      },
    },
  ],
  "_type": "datadog_host",
  "aliases": Array [
    "Dhaiwats-MacBook-Air.local",
  ],
  "apps": Array [
    "ntp",
    "agent",
  ],
  "awsName": undefined,
  "createdOn": undefined,
  "displayName": "Dhaiwats-MacBook-Air.local",
  "hostname": "Dhaiwats-MacBook-Air.local",
  "id": "6264691456",
  "isMuted": false,
  "lastReportedTime": 1657376703,
  "muteTimeout": undefined,
  "name": "Dhaiwats-MacBook-Air.local",
  "sources": Array [
    "agent",
  ],
  "state": "running",
}
`;
